client/request.go

```diff
@@ -263,6 +263,10 @@ func (cli *Client) addHeaders(req *http.Request, headers headers) *http.Request
 	for k, v := range headers {
 		req.Header[http.CanonicalHeaderKey(k)] = v
 	}
+
+	if cli.userAgent != "" {
+		req.Header.Set("User-Agent", cli.userAgent)
```

Comment:
I think  `cli.userAgent` should be `*string` and `*cli.userAgent == ""` should just unset the `User-Agent` header.

Reply:
I was thinking about that, but couldn't come-up with a case where we would first "set" and then "clear" the User-Agent; did you have a specific situation in mind?

Other things I was contemplating;

- Should `WithUserAgent("")` produce an error?
- What user-agent should be set if `*cli.userAgent == ""`? Should we set some custom default? (Would we potentially end-up with Golang's default user-agent?) https://github.com/golang/go/blob/c71cbd544e3da139badd4c03612af41b63711705/src/net/http/request.go#L519-L523

```go
// NOTE: This is not intended to reflect the actual Go version being used.
// It was changed at the time of Go 1.1 release because the former User-Agent
// had ended up blocked by some intrusion detection systems.
// See https://codereview.appspot.com/7532043.
const defaultUserAgent = "Go-http-client/1.1"
```